### PR TITLE
Disable libraw flip 

### PR DESCRIPTION
**Pull request by servantftechnicolor**
Make sure libraw does not reorient our buffer.

A PR is under review at OiiO to give back the flip value even if its application is disabled.

https://github.com/OpenImageIO/oiio/pull/3669

### CODE REVIEW

File: src/aliceVision/image/io.cpp [hunk 1]
```diff
@@ -281,7 +281,10 @@ oiio::ParamValueList readImageMetadata(const std::string& path, int& width, int&
 
 oiio::ImageSpec readImageSpec(const std::string& path)
 {
-  std::unique_ptr<oiio::ImageInput> in(oiio::ImageInput::open(path));
+  oiio::ImageSpec configSpec;
+  configSpec.attribute("raw:user_flip", 1); // set flip to 1 to disable auto rotation of buffer
```
Comment (fabiencastan) [inline src/aliceVision/image/io.cpp:285]:
```suggestion
#if OIIO_VERSION > (10000 * 2 + 100 * 4 + 5) // OIIO_VERSION > 2.4.5
    // To disable the application of the orientation, we need the PR https://github.com/OpenImageIO/oiio/pull/3669,
    // so we can disable the auto orientation and keep the metadata.
    configSpec.attribute("raw:user_flip", 1); // set flip to 1 to disable auto rotation of the image buffer
#endif 
```

File: src/aliceVision/image/io.cpp [hunk 2]
```diff
@@ -372,6 +375,7 @@ void readImage(const std::string& path,
   // use_camera_matrix: Whether to use the embedded color profile, if it is present: 0=never, 1 (default)=only for DNG files, 3=always
   configSpec.attribute("raw:use_camera_matrix", 3); // use embeded color profile
   configSpec.attribute("raw:ColorSpace", "Linear"); // use linear colorspace with sRGB primaries
+  configSpec.attribute("raw:user_flip", 1); // set flip to 1 to disable auto rotation of buffer
```
Comment (fabiencastan) [inline src/aliceVision/image/io.cpp:378]:
```suggestion
#if OIIO_VERSION > (10000 * 2 + 100 * 4 + 5) // OIIO_VERSION > 2.4.5
    // To disable the application of the orientation, we need the PR https://github.com/OpenImageIO/oiio/pull/3669,
    // so we can disable the auto orientation and keep the metadata.
  configSpec.attribute("raw:user_flip", 1); // set flip to 1 to disable auto rotation of buffer
#endif
```

File: src/aliceVision/image/io.cpp [hunk 3]
```diff
@@ -440,6 +443,8 @@ void readImage(const std::string& path,
         // libRAW configuration
         // See https://openimageio.readthedocs.io/en/master/builtinplugins.html#raw-digital-camera-files
 
+        configSpec.attribute("raw:user_flip", 1); // set flip to 1 to disable auto rotation of buffer
```
Comment (fabiencastan) [inline src/aliceVision/image/io.cpp:446]:
```suggestion
#if OIIO_VERSION > (10000 * 2 + 100 * 4 + 5) // OIIO_VERSION > 2.4.5
	    // To disable the application of the orientation, we need the PR https://github.com/OpenImageIO/oiio/pull/3669,
	    // so we can disable the auto orientation and keep the metadata.
        configSpec.attribute("raw:user_flip", 1); // set flip to 1 to disable auto rotation of the image buffer
#endif
```

**Comment by mugulmd** [comment]
Tested with OpenImageIO 2.4.7.0 (contains fix https://github.com/OpenImageIO/oiio/pull/3669) and RAW images with `raw:flip` metadata set to 5.
Results:
- without this PR: buffers are rotated (in my case dimensions should be 6744x4502 but they are 4502x6744) and `Orientation` metadata is 1
- with this PR: buffers are not rotated and `Orientation` metadata is 5.

> Note: there is no need to make a PR in QtAliceVision to report the changes proposed in QtOIIO (cf . https://github.com/alicevision/QtOIIO/pull/26) since QtAliceVision directly calls `aliceVision::image::readImage` and the bug with orientations 5 and 7 has already been fixed

**Comment by mugulmd** [comment]
Tried to create a panorama with this branch using images with `flip` tag 5: 
there is an orientation problem when merging the different viewpoints, which leads to incoherent results.

This orientation problem is already visible at the PanoramaWarping stage, but it may already be present in the previous stages.

File: src/aliceVision/image/io.cpp [hunk 4]
```diff
@@ -352,7 +352,14 @@ oiio::ParamValueList readImageMetadata(const std::string& path, int& width, int&
 
 oiio::ImageSpec readImageSpec(const std::string& path)
 {
-  std::unique_ptr<oiio::ImageInput> in(oiio::ImageInput::open(path));
+  oiio::ImageSpec configSpec;
+#if OIIO_VERSION > (10000 * 2 + 100 * 4 + 5) // OIIO_VERSION > 2.4.5
```
Comment (mugulmd) [inline src/aliceVision/image/io.cpp:356]: the final fix was in version 2.4.12.0 (same remark applies in `readImage`)

**Pull request merged**